code/dynamic_programming/coin_change/coinchange.cpp

```diff
@@ -4,52 +4,33 @@ using namespace std;
 /*
  * Part of Cosmos by OpenGenus Foundation
 */
+#define MAX 100
 
-///Coin Change Problem 
-///Top Down DP
-int dp[100] = {0};
-int coinWaysTD(int amt,int coins[],int n){
-    if(amt==0){
-        dp[0] = 1;
-        return 1;
-    }
-    ///Recursive Case
-    int ans = 0;
-    
-    if(dp[amt]!=0){
-        return dp[amt];
-    }
-    for(int i=0;i<n;i++){
-        if(amt-coins[i]>=0){
-            ans += coinWaysTD(amt-coins[i],coins,n);
-        }
-    }
-    ///Store and return for the first time
-    dp[amt] = ans;
-    return ans;
-}
-
-///Bottom Up DP
-int coinWaysBU(int amt,int coins[],int n){
-    int dp[100] = {0};
-    dp[0] = 1;
+int coinWays(int amt,int coins[],int n){
```

Comment:
Vectors would be the better choice here, since this is C++. Will not need to pass size either for vectors.

Reply:
I just extended whatever code was previously present. Prefer vectors personally. Will do the changes.